cpp/oneapi/dal/algo/knn/backend/gpu/infer_kernel_impl_dpc.hpp

```diff
@@ -473,6 +475,7 @@ sycl::event bf_kernel(sycl::queue& queue,
         distance_impl->get_daal_distance_type() == daal_distance_t::cosine;
     const bool is_euclidean_distance =
         is_minkowski_distance && (distance_impl->get_degree() == 2.0);
+    ONEDAL_ASSERT(is_minkowski_distance ^ is_chebyshev_distance ^ is_cosine_distance);
```

Comment:
```suggestion
    ONEDAL_ASSERT(is_minkowski_distance || is_chebyshev_distance || is_cosine_distance);
```

Reply:
Would ^ make more sense since we want to ensure that exactly one is true?